### PR TITLE
r_d11: Algorithms. Working with cycles|arrays

### PR DESCRIPTION
**Pull request by ShekhaninKyryl**
Please review this PR:

```
1) Реалізувати функцію лінійного пошуку в двовимірному масиві; bool find(int arr_2d[ROWS][COLUMNS], int toFind)

2) Функція bool isSorted(const int* arr, int size, SortingDirection direction) яка перевіряє, чи заданий масив посортований за спаданням, зростанням, де Sorting Direction: enum class SortingDirection { ascending, descending };

3) Написати функції обходу двовимірного масива різними способами:
- StackColsRevers,
- SnakeRowsRevers,
```

### CODE REVIEW

File: r_d11/r_d11.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+
+#define ROWS 3
+#define COLS 3
+
+/*
+1) ���������� ������� �������� ������ � ����������� �����;
```
Comment (ShekhaninKyryl) [inline r_d11/r_d11.cpp]: Sorry forgot to delete

File: r_d11/r_d11.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+
+#define ROWS 3
```
Comment (ShekhaninKyryl) [inline r_d11/r_d11.cpp:3]: When #define is better than const?

Comment (bohdanpc) [reply r_d11/r_d11.cpp:3]: never)

File: r_d11/r_d11.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    if (size <= 0) return false;
+
+    switch (direction)
+    {
+    case SortingDirection::ASCENDING: {
+        int min = INT16_MIN;
+        for (unsigned int i = 0; i < size; i++)
+        {
+            if (min >= arr[i]) return false;
+            min = arr[i];
```
Comment (bohdanpc) [inline r_d11/r_d11.cpp:36]: simply compare arr[i] and arr[i+1]

File: r_d11/r_d11.cpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    if ((distance / (ROWS)) % 2) orientation = -1;
+    pointer += orientation;
+
+    distance = pointer - arr_2d[0];
+    if (orientation == 1 && ((distance / (ROWS)) % 2)) pointer -= (ROWS + 1);
+    if (orientation == -1 && !(distance / (ROWS)) % 2) pointer -= (ROWS - 1);
+
+    return *pointer;
+}
```
Comment (bohdanpc) [inline r_d11/r_d11.cpp:71]: That's some crazy overcomplication for simple array traversal with 2 loops🥲  I don't think I have enough power of will to demystify that, I hope it does work though)

**Pull request merged**